_posts/intjnotinfp/2020-09-02-logical-turtle-post.md

```diff
@@ -0,0 +1,28 @@
++++
+layout: post
+author: intjnotinfp
+title: "Added intjnotinfp's logical turtle"
++++
+Here's the code for my logical turtle:
+<iframe src="https://trinket.io/embed/python/15e54063fd" width="100%" height="600" frameborder="0" marginwidth="0" marginheight="0" allowfullscreen></iframe>
+
+On code:
+I struggled a little bit with brainstorming /executing this exercise in a timely manner and had to come back later to make improvements to my logical turtle exercise. I was able to satisfy most of the requirements in my code (variables, conditionals, and user input) but got really stumped on how to incorporate operators/operands. When it dawned on me that I could "use math" to just build a shape (I initially didn't want to have to conceptualize and go through the hassle of telling the turtle how far, what angle and what direction to go to make a shape), I looked to the internet for some source code on drawing shapes in turtle. For some reference, at this point I'd only made custom polygons that took too long and looked crappy imo. So, it appeared that a hexagon was quick and easy enough, so after I modified the dimensions, I plugged in the following code and ran the program:
+
+polygon = turtle.Turtle()
+my_num_sides = 6
+my_side_length = 70
+my_angle = 360.0 / my_num_sides
+for i in range(my_num_sides):
+   polygon.forward(my_side_length)           
+   polygon.right(my_angle) 
+turtle.done()
```

Comment:
Enclosing code blocks like this in backticks with a language name will make it render prettily on our site. That synatax is:

~~~
```python
# Prettily formatted Python code goes here
```
~~~
This isn't required, but it would enhance your excellent reflection!
Here's how your code would look with that:

```python
polygon = turtle.Turtle()
my_num_sides = 6
my_side_length = 70
my_angle = 360.0 / my_num_sides
for i in range(my_num_sides):
   polygon.forward(my_side_length)           
   polygon.right(my_angle) 
turtle.done()
```